Feat(eos_designs)!: Add support for asdot bgp notation in the design config (#5867)
aristanetworks/ansible-avd: docs/porting-guides/6.x.x.md

Reviewer: ```suggestion
Starting AVD 6.0.0, BGP AS notation can be set by defining `bgp_as_notation` to `auto` (default), `asdot` or `asplain`.
```

Reviewer: ```suggestion
When `bgp_as_notation: auto` (default), AVD takes decision based on the device input AS number:
```

Author: Replaced

      ipv4_acl: IPV4
```

### Configure BGP AS notation to Asdot/Asplain using `bgp_as_notation`

> Till AVD 5.x, All BGP AS would get converted to `asplain` in EOS even when it is given in `asdot` format in eos_designs input, since the option to set AS notation was not available in eos_designs.

Reviewer 2: `Till AVD 5.x` make sound like it includes 4.x but not 5.x?

```suggestion
In AVD 5.x, All BGP AS would get converted to `asplain` format in EOS even when it is given in `asdot` format in eos_designs input, since the option to set AS notation was not available in eos_designs.
```

> Starting AVD 6.0.0, To configure BGP AS notation set `bgp_as_notation: auto/asdot/asplain` default is `auto`.

Reviewer 2: This line may need to be rephrased? Not sure it's really clear.

Author: Changed a bit, please have a look.

> If `bgp_as_notation` is not defined or set to `auto`, it will check its own AS number if dot(.) is found in the input it will convert all the other AS numbers to `asdot` notation and if dot(.) is not found in the input it will convert all the other AS numbers to `asplain` notation.

Reviewer 2: May need to be rephrased as well?

Author: Changed please have a look

      ipv4_acl: IPV4
```

### Configure BGP AS notation to Asdot/Asplain using `bgp_as_notation`

In AVD 5.x, All BGP AS would get converted to `asplain` format in EOS even when it is given in `asdot` format in eos_designs input, since the option to set AS notation was not available in eos_designs.

Starting AVD 6.0.0, BGP AS notation can be set by defining `bgp_as_notation` to `auto`, `asdot` or `asplain` default value is `auto`.

If `bgp_as_notation` is set to `auto` or left undefined, the system examines its own AS number.

- If the AS number contains a dot (`.`), all other AS numbers get converted to `asdot` notation.
- If the AS number does not contain a dot, all other AS numbers get converted to `asplain` notation.

```diff

> + bgp_as_notation: asdot

Reviewer: the porting guide should indicate how to retain the old behavior so probably should say - "to retain old behavior, use `asplain`"

Author: added

Reviewer: the diff needs to show how to retain the old behavior (if this is possible .. chedck comment above)
```suggestion
```diff
+ bgp_as_notation: asplain
```

- If the AS number contains a dot (`.`), all other AS numbers get converted to `asdot` notation.

> - If the AS number does not contain a dot, all other AS numbers get converted to `asplain` notation.

Reviewer: ```suggestion
    - If the AS number contains a dot (`.`), all other AS numbers get converted to `asdot` notation.
    - If the AS number does not contain a dot, all other AS numbers get converted to `asplain` notation.
```

> To retain the old behavior, set `bgp_as_notation` to `asplain`.

Reviewer: actually can you even retain the old behavior? now that I think about it you may have changes no matter what you do if you had a mix and match no?

Author: hmm.. yes, if we have a `.` (dot) in a string (number) and `bgp_as_notation: asplain` then it will convert it. so..  I do not think we can retain the old behavior

      ipv4_acl: IPV4
```

### Configure BGP AS notation to Asdot/Asplain using `bgp_as_notation`

> In AVD 5.x, All BGP AS would get converted to `asplain` format in EOS even when it is given in `asdot` format in eos_designs input, since the option to set AS notation was not available in eos_designs.

Reviewer 2: ```suggestion
In AVD 5.x, all BGP AS would get converted to `asplain` format in EOS even when it is given in `asdot` format in eos_designs input, since the option to set AS notation was not available in eos_designs.
```

Starting AVD 6.0.0, BGP AS notation can be set by defining `bgp_as_notation` to `auto`, `asdot` or `asplain` default value is `auto`.

When `bgp_as_notation: auto` (default), AVD takes decision based on the device input AS number:
    - If the AS number contains a dot (`.`), all other AS numbers get converted to `asdot` notation.
    - If the AS number does not contain a dot, all other AS numbers get converted to `asplain` notation.

> To retain the old behavior, set `bgp_as_notation` to `asplain`.

Reviewer: ```suggestion
It may not be possible to retain the old behavior if you had a mix and match of asdot and asplain notation, which was not rendering what EOS would have rendered anyway.
```

```diff
+ bgp_as_notation: asplain

l3leaf:
  defaults:
    bgp_as: 1.1

>  ```

Reviewer: ```suggestion
```

    - If the AS number contains a dot (`.`), all other AS numbers get converted to `asdot` notation.
    - If the AS number does not contain a dot, all other AS numbers get converted to `asplain` notation.

> It may not be possible to retain the old behavior if you had a mix and match of asdot and asplain notation, which was not rendering what EOS would have rendered anyway.

Reviewer: @ClausHolbechArista ok with the wording?
